connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -24,24 +24,32 @@ import scala.collection.JavaConverters._
  */
 class UCSingleCatalog extends TableCatalog with SupportsNamespaces {
 
-  private var deltaCatalog: TableCatalog = null
+  @volatile private var delegate: TableCatalog = null
 
   override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
     val proxy = new UCProxy()
     proxy.initialize(name, options)
-    deltaCatalog = Class.forName("org.apache.spark.sql.delta.catalog.DeltaCatalog")
-      .getDeclaredConstructor().newInstance().asInstanceOf[TableCatalog]
-    deltaCatalog.asInstanceOf[DelegatingCatalogExtension].setDelegateCatalog(proxy)
+    try {
+      if (options.containsKey("__TEST_NO_DELTA__")) {
```

Comment:
cc @tdas for this discussion: https://github.com/unitycatalog/unitycatalog/pull/467#discussion_r1755425624

Reply:
I am okay with making class name configurable. there might be a simpler approach though. have 2 static threadlocal mutable boolean variable. 
1. one dictates whether to load delta catalog or not
2. other is set if the deltacatalog loaded or not
with that you can set the first variable to load / not load the DeltaCatalog (and reset the second one to false), and then test whether DeltaCatalog was loaded or not.